### PR TITLE
Adding pagination to some GET endpoints

### PR DESCRIPTION
**Pull request by LanternNassi**
# Description

Adding pagination to relevant critical GET endpoints 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Trello Ticket ID

https://trello.com/c/EL7nv3ce/1375-add-pagination-to-most-endpoints

## How Can This Be Tested?

This can be tested by adding page and per_page values as query parameters with their respective values to the modified GET endpoint URL

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/controllers/users.py [hunk 1]
```diff
@@ -120,17 +120,19 @@ def get(self):
...
-        users_data, errors = user_schema.dumps(users)
+        users_data, errors = user_schema.dumps(users.items)
 
         if errors:
             return dict(status='fail', message=errors), 400
 
         return dict(
             status='success',
-            data=dict(users=json.loads(users_data))
+            data=dict(users=json.loads(users_data) , pagination=users.pagination)
```
Comment (Mubangizi) [inline app/controllers/users.py:135]:
```suggestion
            data=dict(pagination=users.pagination, users=json.loads(users_data))
```

File: app/controllers/transactions.py [hunk 2]
```diff
@@ -148,13 +151,13 @@ def get(self, project_id):
...
 
-        transaction_data, errors = transaction_schema.dumps(transaction)
+        transaction_data, errors = transaction_schema.dumps(transaction.items)
 
         if errors:
             return dict(status='fail', message=errors), 500
 
         return dict(status='success', data=dict(
-            transaction=json.loads(transaction_data))), 200
+            transaction=json.loads(transaction_data), pagination=transaction.pagination)), 200
```
Comment (Mubangizi) [inline app/controllers/transactions.py:160]:
```suggestion
           pagination=transaction.pagination, transaction=json.loads(transaction_data))), 200
```

File: app/controllers/receipts.py [hunk 3]
```diff
@@ -30,21 +33,21 @@ def get(self, project_id):
...
 
-        billing_receipts_data, errors = billing_receipt_schema.dumps(billing_receipts)
+        billing_receipts_data, errors = billing_receipt_schema.dumps(billing_receipts.items)
 
         if errors:
             return dict(status='fail', message=errors), 500
 
         return dict(status='success', data=dict(
-            billing_receipts=json.loads(billing_receipts_data))), 200
+            billing_receipts=json.loads(billing_receipts_data) , pagination=billing_receipts.pagination)), 200
```
Comment (Mubangizi) [inline app/controllers/receipts.py:50]: Same here
```suggestion
            pagination=billing_receipts.pagination, billing_receipts=json.loads(billing_receipts_data))), 200
```

File: app/controllers/project_database.py [hunk 4]
```diff
@@ -645,16 +649,20 @@ def get(self):
...
+        database_data, errors = database_schema.dumps(databases.items)
 
         if errors:
             return dict(status='fail', message=errors), 500
 
         database_data_list = json.loads(database_data)
+        pagination = databases.pagination
 
-        return dict(status='success', data=dict(databases=database_data_list)), 200
+        return dict(status='success', data=dict(databases=database_data_list, pagination=pagination)), 200
```
Comment (Mubangizi) [inline app/controllers/project_database.py:665]:
```suggestion
        return dict(status='success', pagination=pagination, data=dict(databases=database_data_list)), 200
```

File: app/controllers/project_database.py [hunk 5]
```diff
@@ -213,7 +217,7 @@ def get(self, project_id):
 
             database['db_status'] = db_status
 
-        return dict(status='success', data=dict(databases=database_data_list)), 200
+        return dict(status='success', data=dict(databases=database_data_list, pagination=pagination)), 200
```
Comment (Mubangizi) [inline app/controllers/project_database.py:220]:
```suggestion
        return dict(status='success', pagination=pagination, data=dict(databases=database_data_list)), 200
```

File: app/controllers/billing_invoice.py [hunk 6]
```diff
@@ -164,13 +167,13 @@ def get(self, project_id):
...
         billing_invoice_data, errors = billing_invoice_schema.dumps(
-            billing_invoice)
+            billing_invoice.items)
 
         if errors:
             return dict(status='fail', message=errors), 500
 
         return dict(status='success', data=dict(
-            billing_invoice=json.loads(billing_invoice_data))), 200
+            billing_invoice=json.loads(billing_invoice_data), pagination=billing_invoice.pagination)), 200
```
Comment (Mubangizi) [inline app/controllers/billing_invoice.py:176]:
```suggestion
pagination=billing_invoice.pagination, billing_invoice=json.loads(billing_invoice_data))), 200
```

**Review by Mubangizi — CHANGES_REQUESTED** [review]
Hey @LanternNassi, Great work on this. 
Just a quick suggestion for you to put `Pagination` before the `items` list data, such that when anyone is reading through the data, they don't have to scroll at the bottom of the list to see the pagination info

**Review by Mubangizi — APPROVED** [review]
LGTM

**Review by MutegekiHenry — APPROVED** [review]
LGTM

**Pull request merged**